webapp/src/features/adminNotificationTab/AdminNotificationTab.tsx

```diff
@@ -0,0 +1,242 @@
+import { Action } from '@material-table/core'
+import { Route, Routes, useLocation, useNavigate } from 'react-router-dom'
+import { NotFound } from '../../pages/404'
+import AdminEditNotification from '../adminEditNotification/AdminEditNotification'
+import AdminAddNotification from '../adminAddNotification/AdminAddNotification'
+import { AdminEmailNotification } from '../../models/Notifications'
+
+const getTitle = (activeTab: string) => {
+  if (activeTab === undefined) {
+    return 'CV Manager Email Notifications'
+  } else if (activeTab === 'editNotification') {
+    return 'Edit Email Notification'
+  } else if (activeTab === 'addNotification') {
+    return 'Add Email Notification'
+  }
+  return 'Unknown'
+}
+
+const AdminNotificationTab = () => {
+  const dispatch: ThunkDispatch<RootState, void, AnyAction> = useDispatch()
+  const navigate = useNavigate()
+  const location = useLocation()
+
+  const activeTab = location.pathname.split('/')[3]
+  const title = getTitle(activeTab)
+
+  const tableData = useSelector(selectTableData)
+  const [columns] = useState([
+    { title: 'First Name', field: 'first_name', id: 0 },
+    { title: 'Last Name', field: 'last_name', id: 1 },
+    { title: 'Email', field: 'email', id: 2 },
+    { title: 'Email Type', field: 'email_type', id: 3 },
+  ])
+  const loading = useSelector(selectLoading)
+
+  let tableActions: Action<AdminEmailNotification>[] = [
+    {
+      icon: 'delete',
+      tooltip: 'Delete Email Notification',
+      position: 'row',
+      onClick: (event, rowData: AdminEmailNotification) => {
+        const buttons = [
+          {
+            label: 'Yes',
+            onClick: () => dispatch(deleteNotifications([rowData])),
+          },
+          {
+            label: 'No',
+            onClick: () => {},
+          },
+        ]
+        const alertOptions = Options(
+          'Delete Email Notification',
+          'Are you sure you want to delete ' + rowData.email_type + ' for "' + rowData.email + '"?',
+          buttons
+        )
+        confirmAlert(alertOptions)
+      },
+    },
+    {
+      icon: 'edit',
+      tooltip: 'Edit Notification',
+      position: 'row',
+      onClick: (event, rowData: AdminEmailNotification) => onEdit(rowData),
+    },
+    {
+      tooltip: 'Remove All Selected Notifications',
+      icon: 'delete',
+      onClick: (event, rowData: AdminEmailNotification[]) => {
+        const buttons = [
+          {
+            label: 'Yes',
+            onClick: () => dispatch(deleteNotifications(rowData)),
+          },
+          {
+            label: 'No',
+            onClick: () => {},
+          },
+        ]
+        const alertOptions = Options(
+          'Delete Selected Email Notifications',
+          'Are you sure you want to delete ' + rowData.length + ' email notifications?',
+          buttons
+        )
+        confirmAlert(alertOptions)
+      },
+    },
+  ]
+
+  // dispatch
+  const updateTableData = async () => {
```

Comment:
This does not appear to be called when refreshing onto this page directly. Haven't dug through the logic yet. but refreshing the webapp while on this page, doesn't pull initial data

Reply:
I added a useEffect call to pull data on the first render - this call is only used when the refresh button is pressed.